src/assets/Keycloak/Keycloak.md

```diff
@@ -0,0 +1,147 @@
+
+3. Standard Protocols:
+   Keycloak is based on three standard protocols i.e., OpenID Connect, OAuth2.0, and SAML.
+
+4. Password Policies:
+   Keycloak contains different password policies i.e., HashAlgorithm, Hashing Iterations, Digits, Lowercase Characters, Regular Expression and so on.
+
+5. Single Sign-On:
+   Keycloak facilitates a seamless login experience where users only need to sign in once to access multiple applications, eliminating the need for repeated logins.
+   This not only streamlines user experience but also simplifies identity management for administrators.
+
+6. Social Login:
+   Keycloak has built-in support to login via Google, GitHub, and Facebook, which helps us to use social identity providers.
+
+## Keycloak Drawbacks
+Despite its extensive features, Keycloak does have certain limitations. One of these is the need for a more varied implementation approach.
+Additionally, the following are some of the drawbacks associated with Keycloak:
+1. Complex Server Deployment: The manual process involved in deploying Keycloak on a server can be complicated, potentially affecting overall productivity.
+2. Lack of Comprehensive Support Documentation: Despite its robust functionalities, Keycloak could benefit from more efficient and extensive support documentation. This would make it easier for users to find suitable solutions to their problems.
+
+## Keycloak Installation
+### By Docker image
+Before we start, lets make sure following things get installed into your system.
+- install [Docker Engine](https://docs.docker.com/engine/install/)
+- install [Docker Compose](https://docs.docker.com/compose/install/)
+
+1. Keycloak using docker image
+Run the following docker command to serve the Keycloak image. Keycloak will be available on `http://localhost:8080.
+```bash
+docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:24.0.2 start-dev
+ ```
+
+2. Keycloak with postgres database using docker-compose
+```
+services:
+  postgres:
+    image: postgres:alpine
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_DB: keycloak
+      POSTGRES_USER: keycloak
+      POSTGRES_PASSWORD: keycloak
+
+  keycloak:
+    image: quay.io/keycloak/keycloak:24.0.1
+    command: ["start-dev"]
+    environment:
+      KC_DB: postgres
+      KC_DB_URL: "jdbc:postgresql://postgres:5432/keycloak"
+      KC_DB_USERNAME: keycloak
+      KC_DB_PASSWORD: keycloak
+      KC_FEATURES: impersonation
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: admin
+    ports:
+      - 8001:8080 # map keycloak port 8080 to 8001
+    depends_on:
+      - postgres
+
+volumes:
+  postgres_data:
+```
+
+Save above code in `compose.yaml` file and run `docker compose up` in terminal.
+
+Docker compose serves Keycloak on `http://localhost:8001`
```

Comment:
```suggestion
Docker compose serves Keycloak on `http://localhost:8001`.
```

Reply:
In English, there should be a point at the end of each full sentence. If you think it's confusing because this is a URL, then at least apply the same writing style to the whole blog post, for example line 56...